client/QuickStart/SampleApp.cs

```
@@ -0,0 +1,303 @@
+
+                // Tip: This is generally a one-time configuration. Learn More: For more information about providing inline mappings and mapping references,
+                // see: https://docs.microsoft.com/azure/data-explorer/kusto/management/mappings
+                if (!await CreateIngestionMappingsAsync(bool.Parse(file["useExistingMapping"]), adminClient, config.DatabaseName, config.TableName, mappingName, file["mappingValue"], dataFormat))
+                    continue;
+
+                // Learn More: For more information about ingesting data to Kusto in C#,
+                // see: https://docs.microsoft.com/en-us/azure/data-explorer/net-sdk-ingest-data
+                await IngestDataAsync(file, dataFormat, ingestClient, config.DatabaseName, config.TableName, mappingName);
+            }
+
+            await Util.WaitForIngestionToCompleteAsync(config.WaitForIngestSeconds);
+        }
+
+        /// <summary>
+        /// Creates Ingestion Mappings (if required) based on given values.
+        /// </summary>
+        /// <param name="useExistingMapping">Flag noting if we should the existing mapping or create a new one</param>
+        /// <param name="adminClient">Privileged client to run Control Commands</param>
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="configTableName">Table name</param>
+        /// <param name="mappingName">Desired mapping name</param>
+        /// <param name="mappingValue">Values of the new mappings to create</param>
+        /// <param name="dataFormat">Given data format</param>
+        /// <returns>True if Ingestion Mappings exists (whether by us, or the already existing one)</returns>
+        private static async Task<bool> CreateIngestionMappingsAsync(bool useExistingMapping, ICslAdminProvider adminClient, string configDatabaseName, string configTableName, string mappingName, string mappingValue, DataSourceFormat dataFormat)
```

Why do you completely swallow the exception here? even if it's valid to continue after, at least show what it was

---

Utils.ExecuteAsync which is called there handles the exception internally, that catch block is there only due to syntactical requirements. perhaps i should just remove the try catch block here. WDYT?

---

Yes